shopinvader_invoice/tests/common.py

```diff
@@ -74,6 +74,42 @@ def _check_data_content(self, data, invoices):
                 invoice.amount_total_signed,
             )
             self.assertEqual(current_data.get("amount_due"), invoice.amount_residual)
+
+            self.assertEqual(
+                len(current_data.get("lines")), len(invoice.invoice_line_ids)
+            )
+
+            # lines are not ordered (sorted)
+            # so this code checks each line of the invoice
+            # with each line of the output
+            number_of_lines_ok = 0
+            for current_line in current_data.get("lines"):
+                for line in invoice.invoice_line_ids:
```

Comment:
The nested loop is not correct/logic.
As the order should be the same, you can use: `for current_line, line in zip(current_data.get("lines"), invoice.invoice_line_ids):`

Reply:
And tests are green because you probably have only 1 invoice line. It'll fail with more than 1 line. Or maybe I'm missing something.

Reply:
>As the order should be the same

I don't want to force an order on the lines.

I added a second invoice line in one of the tests

Reply:
Maybe I'm missing something but how it could be equals?

You have `[{...}, {...}, ...]` (who is `current_data.get("lines")`). Then for each element of this list, you loop on each invoice line (`invoice.invoice_line_ids`).
So the first dict match with the first invoice line (it's ok), but in the second iteration of the nested loop, you compare the first dict with the second invoice line. Your test should fail at this point.

Reply:
Indeed the test fails ``` n²-n``` times and succeed `n` times.